rasterio/_base.pyx

```diff
@@ -921,6 +924,22 @@ cdef class DatasetBase(object):
                 self._gcps = self.get_gcps()
             return self._gcps
 
+    property files:
+
+        """Returns a sequence of files associated with the dataset.
+
+        Returns
+        -------
+        tuple
+        """
+
+        def __get__(self):
+            cdef GDALDatasetH h_dataset = NULL
+            h_dataset = self.handle()
+            with nogil:
+                file_list = GDALGetFileList(h_dataset)
+            num_items = CSLCount(file_list)
+            return tuple([file_list[i] for i in range(num_items)])
```

Comment:
@geowurster I think we need a try/finally in here with `CSLDestroy(file_list)` called when we finalize. According to http://www.gdal.org/classGDALDataset.html#aefe21eb2d1c87304a14f81ee2e6d9717, the file list is owned by the caller.